m2m-oauth-server/test/test.go

```diff
@@ -0,0 +1,356 @@
+	RequireDeviceID:     false,
+	RequireOwner:        true,
+	AccessTokenLifetime: 0,
+	AllowedGrantTypes:   []service.GrantType{service.GrantTypeClientCredentials},
+	AllowedAudiences:    nil,
+	AllowedScopes:       nil,
+	JWTPrivateKey: service.PrivateKeyJWTConfig{
+		Enabled:       true,
+		Authorization: config.MakeAuthorizationConfig(),
+	},
+}
+
+var DeviceProvisioningServiceOAuthClient = service.Client{
+	ID:                  "deviceProvisioningServiceClient",
+	SecretFile:          "data:,deviceProvisioningServiceClientSecret",
+	RequireDeviceID:     true,
+	RequireOwner:        true,
+	AccessTokenLifetime: 0,
+	AllowedGrantTypes:   []service.GrantType{service.GrantTypeClientCredentials},
+	AllowedAudiences:    nil,
+	AllowedScopes:       nil,
+}
+
+var OAuthClients = service.OAuthClientsConfig{
+	&ServiceOAuthClient,
+	&JWTPrivateKeyOAuthClient,
+	&DeviceProvisioningServiceOAuthClient,
+}
+
+func MakeConfig(t require.TestingT) service.Config {
+	var cfg service.Config
+
+	cfg.Log = log.MakeDefaultConfig()
+
+	cfg.APIs.HTTP.Connection = config.MakeListenerConfig(config.M2M_OAUTH_SERVER_HTTP_HOST)
+	cfg.APIs.HTTP.Connection.TLS.ClientCertificateRequired = false
+	cfg.APIs.HTTP.Server = config.MakeHttpServerConfig()
+	cfg.Clients.OpenTelemetryCollector = kitNetHttp.OpenTelemetryCollectorConfig{
+		Config: config.MakeOpenTelemetryCollectorClient(),
+	}
+
+	cfg.OAuthSigner.PrivateKeyFile = urischeme.URIScheme(os.Getenv("M2M_OAUTH_SERVER_PRIVATE_KEY"))
+	cfg.OAuthSigner.Domain = config.M2M_OAUTH_SERVER_HTTP_HOST
+	cfg.OAuthSigner.Clients = OAuthClients
+	cfg.OAuthSigner.OwnerClaim = OwnerClaim
+	cfg.OAuthSigner.DeviceIDClaim = DeviceIDClaim
+	err := cfg.Validate()
+	require.NoError(t, err)
+
+	return cfg
+}
+
+func SetUp(t require.TestingT) (tearDown func()) {
+	return New(t, MakeConfig(t))
+}
+
+func New(t require.TestingT, cfg service.Config) func() {
+	ctx := context.Background()
+	logger := log.NewLogger(cfg.Log)
+
+	fileWatcher, err := fsnotify.NewWatcher(logger)
+	require.NoError(t, err)
+
+	s, err := service.New(ctx, cfg, fileWatcher, logger)
+	require.NoError(t, err)
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		_ = s.Serve()
+	}()
+	return func() {
+		_ = s.Close()
+		wg.Wait()
+		err = fileWatcher.Close()
+		require.NoError(t, err)
+	}
+}
+
+func GetSecret(t require.TestingT, clientID string) string {
+	for _, c := range OAuthClients {
+		if c.ID == clientID {
+			data, err := c.SecretFile.Read()
+			require.NoError(t, err)
+			return string(data)
+		}
+	}
+	require.FailNow(t, "client not found")
+	return ""
+}
+
+func NewRequestBuilder(method, host, url string, body io.Reader) *RequestBuilder {
+	b := RequestBuilder{
+		method:      method,
+		body:        body,
+		uri:         fmt.Sprintf("https://%s%s", host, url),
+		uriParams:   make(map[string]interface{}),
+		header:      make(map[string]string),
+		queryParams: make(map[string]string),
+	}
+	return &b
+}
+
+type RequestBuilder struct {
```

Comment:
This will be the 4th RequestBuilder in the repository, can't the most general one in test/http be used here?

Reply:
using test/http

Reply:
rm?

Reply:
removed